File: VehicleTracker.StorageService/Startup.cs
using System;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using VehicleTracker.StorageService.Storage;
using VehicleTracker.StorageService.Context;
using Microsoft.EntityFrameworkCore;
using Autofac;
using VehicleTracker.StorageService.Model;

namespace VehicleTracker.StorageService
{
    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddOptions();
            services.AddDbContext<VehicleContext>(options =>
                options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
            services.AddMvc();
        }

        public void ConfigureContainer(ContainerBuilder builder)
        {
            builder.RegisterType<VehicleRepository>()
                .As<IRepository<string, Vehicle, VehicleFilter>>()
                .InstancePerLifetimeScope();

            builder.RegisterType<CustomerRepository>()
                .As<IRepository<Guid, Customer, CustomerFilter>>()
                .InstancePerLifetimeScope();
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            app.UseMvc();
        }
    }
}

File: VehicleTracker.StorageService.FSharp/src/VehicleTracker.StorageService.FSharp/Startup.fs
module VehicleTracker.StorageService.FSharp.Startup

open System
open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Cors.Infrastructure
open Microsoft.AspNetCore.Hosting
open Microsoft.Extensions.Logging
open Microsoft.Extensions.DependencyInjection
open Giraffe
open VehicleTracker.StorageService.FSharp.HttpHandlers
open VehicleTracker.StorageService.FSharp.Models
open Microsoft.EntityFrameworkCore
open VehicleTracker.StorageService.FSharp.DataAccess.Customers
open VehicleTracker.StorageService.FSharp.DataAccess.Vehicles
open VehicleTracker.StorageService.FSharp.Routing
open Microsoft.Extensions.Configuration

// ---------------------------------
// Web app
// ---------------------------------
let webApp =
    choose [
        subRoute "/vehicles"
            (choose [
                GET >=> routefs (handleGet getVehicle)
                DELETE >=> routefs (handleDelete deleteVehicle)
                PUT >=> routefs (handlePut updateVehicle)
                GET >=> route "/" >=> handleList listVehicles
                POST >=> route "/" >=> handlePost addVehicle
            ])
        subRoute "/customers"
            (choose [
                GET >=> routefo (handleGet getCustomer)
                DELETE >=> routefo (handleDelete deleteCustomer)
                PUT >=> routefo (handlePut updateCustomer)
                GET >=> route "/" >=> handleList listCustomers
                POST >=> route "/" >=> handlePost addCustomer
            ])                
        setStatusCode 404 >=> text "Not Found" ]

// ---------------------------------
// Error handler
// ---------------------------------
let errorHandler (ex : Exception) (logger : ILogger) =
    logger.LogError(EventId(), ex, "An unhandled exception has occurred while executing the request.")
    clearResponse >=> setStatusCode 500 >=> text ex.Message

// ---------------------------------
// Config
// ---------------------------------
let configureCors (builder : CorsPolicyBuilder) =
    builder.WithOrigins("http://localhost:8093")
           .AllowAnyMethod()
           .AllowAnyHeader()
           |> ignore

let configureApp (app : IApplicationBuilder) =
    let env = app.ApplicationServices.GetService<IHostingEnvironment>()
    (match env.IsDevelopment() with
    | true  -> app.UseGiraffeErrorHandler errorHandler //app.UseDeveloperExceptionPage()
    | false -> app.UseGiraffeErrorHandler errorHandler)
        .UseCors(configureCors)
        .UseGiraffe(webApp)

let configureServices (context : WebHostBuilderContext)  (services : IServiceCollection) =
    services.AddCors()    |> ignore
    services.AddGiraffe() |> ignore
    
    services.AddDbContext<VehiclesContext>(
        fun (options : DbContextOptionsBuilder) -> 
            options.UseSqlServer(context.Configuration.GetConnectionString("DefaultConnection")) |> ignore
        ) |> ignore

    